modules/01-mods.js

```diff
@@ -1,538 +1,434 @@
-    + `**${u.escapeText(interaction.member.displayName)}** has issued you a warning for:\n`
-    + reason;
-  await member.send(response).catch(() => blocked(member));
-
-  const embed = u.embed({ author: member })
-    .setColor("#0000FF")
-    .setDescription(reason)
-    .addFields({ name: "Resolved", value: `${u.escapeText(interaction.user.username)} issued a ${value} point warning.` })
-    .setTimestamp();
-  const flag = await interaction.guild.channels.cache.get(u.sf.channels.modlogs).send({ embeds: [embed] });
-
-  await u.db.infraction.save({
-    discordId: member.id,
-    value: value,
-    description: reason,
-    message: interaction.id,
-    flag: flag.id,
-    channel: interaction.channel.id,
-    mod: interaction.member.id
-  });
-
-  const summary = await u.db.infraction.getSummary(member.id);
-  embed.addFields({ name: `Infraction Summary (${summary.time} Days) `, value: `Infractions: ${summary.count}\nPoints: ${summary.points}` });
-
-  flag.edit({ embeds: [embed] });
-  await interaction.editReply(`${member} has been warned **${value}** points for reason \`${reason}\``);
+  const reason = interaction.options.getString("reason", true);
+  const value = interaction.options.getInteger("value") ?? 1;
+  if (!member) return interaction.editReply(noTarget);
+
+  const warn = await c.warn(interaction, reason, value, member);
+  return interaction.editReply(warn);
 }
 
+/** @param {Augur.GuildInteraction<"CommandSlash">} interaction*/
+async function slashModGrownups(interaction) {
+  const time = Math.min(30, interaction.options.getInteger("time") ?? 15);
+  if (!interaction.channel) return interaction.reply({ content: "Well that's awkward, I can't access the channel you're in!", ephemeral: true });
+  if ((interaction.channel.parentId || "") != u.sf.channels.staffCategory) return interaction.reply({ content: "This command can only be used in the LDSG-Staff Category!", ephemeral: true });
+  interaction.reply(time == 0 ? `*Whistles and wanders back in*` : `*Whistles and wanders off for ${time} minutes...*`);
+
+  if (c.grownups.has(interaction.channel.id)) {
+    clearTimeout(c.grownups.get(interaction.channel.id));
+  }
+  if (time == 0) {
+    c.grownups.delete(interaction.channel.id);
+  } else {
+    c.grownups.set(interaction.channel.id, setTimeout((channel) => {
+      c.grownups.delete(channel.id);
+      channel.send("*I'm watching you again...* :eyes:");
+    }, time * 60 * 1000, interaction.channel));
+  }
+}
+
+
 Module.addInteraction({
   name: "mod",
-  guild: u.sf.ldsg,
+  guildId: u.sf.ldsg,
   id: u.sf.commands.slashMod,
-  permissions: p.isMod,
-  /** @param {Discord.CommandInteraction} interaction*/
-  process: async (interaction) => {
+  onlyGuild: true,
+  permissions: (int) => p.calc(int.member, ["mod"]),
```

Comment:
Argument of type 'GuildMember | APIInteractionGuildMember | null' is not assignable to parameter of type 'GuildMember'.
  Type 'null' is not assignable to type 'GuildMember'.ts(2345)